pkg/mount-manager/safe-mounter_windows.go

```diff
@@ -142,30 +144,37 @@ func (mounter *CSIProxyMounter) Unmount(target string) error {
 }
 
 func (mounter *CSIProxyMounter) GetDevicePath(deviceName string, partition string, volumeKey string) (string, error) {
-	getDiskNumberRequest := &diskv1alpha1.GetDiskNumberByNameRequest{
-		DiskName: deviceName,
-	}
-	getDiskNumberResponse, err := mounter.DiskClient.GetDiskNumberByName(context.Background(), getDiskNumberRequest)
+	id := "page83"
+	listRequest := &diskv1beta1.ListDiskIDsRequest{}
+	diskIDsResponse, err := mounter.DiskClient.ListDiskIDs(context.Background(), listRequest)
 	if err != nil {
 		return "", err
 	}
-	return getDiskNumberResponse.DiskNumber, nil
+	diskIDsMap := diskIDsResponse.GetDiskIDs()
+	for diskNum, diskInfo := range diskIDsMap {
+		klog.V(4).Infof("found disknum %s, diskInfo %v", diskNum, diskInfo)
+		idValue, found := diskInfo.Identifiers[id]
+		if found && strings.Contains(idValue, deviceName) {
```

Comment:
Can you add a comment with an example format of idValue and deviceName?

`strings.Contains` is prone to substring matching. Do we have that issue here?

Reply:
I added comment. From the tests, it seems working fine.

Reply:
What if the idValue is something like "pvc-123", but the device name is "pvc-12"?

Reply:
good point. I should use exact match